Source/Core/Core/IOS/Network/SSL.cpp

```diff
@@ -445,7 +477,7 @@ IPCCommandResult NetSSL::IOCtlV(const IOCtlVRequest& request)
       WiiSockMan& sm = WiiSockMan::GetInstance();
       ssl->hostfd = sm.GetHostSocket(ssl->sockfd);
       INFO_LOG_FMT(IOS_SSL, "IOCTLV_NET_SSL_CONNECT socket = {}", ssl->sockfd);
-      mbedtls_ssl_set_bio(&ssl->ctx, &ssl->hostfd, mbedtls_net_send, mbedtls_net_recv, nullptr);
+      mbedtls_ssl_set_bio(&ssl->ctx, &ssl->hostfd, SSLSendWithoutSNI, mbedtls_net_recv, nullptr);
```

Comment:
The context/userdata pointer (second argument) should be `&ssl->ctx`, not `&ssl->hostfd`.

Better yet, I'd recommend passing `ssl` directly so you don't have to look up the mbedtls SSL context manually.